languages/go/exercises/concept/maps/.meta/example.go

```diff
@@ -0,0 +1,61 @@
+package maps
+
+// units store the Dozen Store unit measurement
+var units = map[string]int{
+	"quarter_of_a_dozen": 3,
+	"half_of_a_dozen":    6,
+	"dozen":              12,
+	"small_gross":        120,
+	"gross":              144,
+	"great_gross":        1728,
+}
+
+// bill to store the item that customer wants to buy
+var bill = map[string]int{}
+
+// AddItem add item to customer bill
+func AddItem(item string, unit string) bool {
+	if _, ok := units[unit]; !ok {
+		return false
+	}
+
+	bill[item] += units[unit]
+
+	return true
+}
+
+// RemoveItem remove item from customer bill
+func RemoveItem(item string, unit string) bool {
+	if _, ok := bill[item]; !ok {
+		return false
+	}
+
+	if _, ok := units[unit]; !ok {
+		return false
+	}
+
+	newUnit := bill[item] - units[unit]
+	if newUnit < 0 {
+		return false
+	} else if newUnit == 0 {
+		delete(bill, item)
+	} else {
+		bill[item] -= units[unit]
+	}
+
+	return true
+}
+
+// Checkout finish the transaction, reset the bill to zero
+func Checkout() {
+	bill = make(map[string]int)
+}
```

Comment:
This function would be something like `NewBill() map[string]int`. I'd move it up to be the first function.

Reply:
> I'd move it up to be the first function.

Sorry, I don't understand, what do you mean by first function?

> This function would be something like NewBill() map[string]int

Sure, I'll probably continue the work on Sunday. I also think the after and introduction files need a some touches

Reply:
> Sorry, I don't understand, what do you mean by first function?

Just to move it to the top of the file. That way the student will start with implementing the `NewBill` function -- which makes probably most sense.

Reply:
I fixit, see a473bdc